Library/Homebrew/utils/bottles.rb

```diff
@@ -75,7 +75,7 @@ def self.package(formula_name)
         package_name = formula_name.to_s.dup
         package_name.tr!("+", "x")
         package_name.sub!(/(.)@(\d)/, "\\1:\\2") # Handle foo@1.2 style formulae.
-        package_name
+        ERB::Util.url_encode(package_name)
```

Comment:
I suspect this actually fixes some of the hacks above like `+` (which would have been interpreted as a space if not URL encoded). Too late to remove them now though.